agora/Agora/Stake/Scripts.hs

```diff
@@ -301,18 +302,40 @@ stakeValidator stake =
           tcassert "Proposal ST spent" $
             spentProposalST #== 1
 
+          -- Update the stake datum, but only the 'lockedBy' field.
+
+          let -- We actually don't know whether the given lock is valid or not.
+              -- This is checked in the proposal validator.
+              newLock = pfield @"lock" # l
+              -- Prepend the new lock to the existing locks.
+              expectedLocks = pcons # newLock # stakeDatum.lockedBy
+
+          expectedDatum <-
+            tclet $
+              pdata $
+                mkRecordConstr
+                  PStakeDatum
+                  ( #stakedAmount .= stakeDatum.stakedAmount
+                      .& #owner .= stakeDatum.owner
+                      .& #lockedBy .= pdata expectedLocks
+                  )
+
           tcassert "A UTXO must exist with the correct output" $
+            -- FIXME: no need to pass the whole txInfo to 'anyOutput'.
             anyOutput @PStakeDatum # txInfo
               #$ plam
               $ \value address newStakeDatum' ->
                 let isScriptAddress = pdata address #== ownAddress
-                    _correctOutputDatum = pdata newStakeDatum' #== pdata stakeDatum'
+                    correctOutputDatum = pdata newStakeDatum' #== expectedDatum
+                    -- TODO: Is this correct? I think We only need to ensure
+                    --       correct amount of GT/SST in the continuing output.
```

Comment:
Technically speaking you're right. But I don't think we save a lot by doing that as opposed to this?

Reply:
If we want to add minimum ada in the stake output in the sample, we'll have to change this. However the sample doesn't reflect the real on-chain environment, so this may not be very important. I'll just mention this in the sample.

Reply:
Hmmm, we definitely need to include minADA in the sample though. Why does this check break that? The important thing is that the minADA already has to exist at that output, but other than that it should be okay, right?

Reply:
> Why does this check break that?

In the sample, the stake input doesn't have any ada, if the stake output have min ada, the check won't pass.

Check [this](https://github.com/Liqwid-Labs/agora/blob/connor/proposal-voting/agora-sample/Sample/Proposal.hs#L321-L336) and [this](https://github.com/Liqwid-Labs/agora/blob/connor/proposal-voting/agora-sample/Sample/Proposal.hs#L366-L381).

If we do want to include min ada in the output without altering the validation logic, the solutuion would be adding min ada to the stake input value.

Reply:
Let's do that.